dbux-data/src/indexes/CollectionIndex.js

```diff
@@ -1,6 +1,85 @@
 import { newLogger } from '@dbux/common/src/log/logger';
 import EmptyObject from '@dbux/common/src/util/EmptyObject';
 
+
+const ArrayIndexManager = {
+  createNew() {
+    return [];
+  },
+  get(key) {
+    return this._byKey[key];
+  },
+  set(key, container) {
+    this._byKey[key] = container;
+  },
+  getOrCreateContainer(key) {
+    let container = this.get(key);
+    if (!container) {
+      container = this._containerMethods.createNewContainer();
```

Comment:
`this._containerMethods` -> `this.manager._containerMethods`?

Also, have you tested this case somehow?
How can we test this case easily?

Reply:
A simple way to test is use break point anywhere that has variable `dp`, and check the `_byKey` in each type of indexes.